src/dmd/dsymbolsem.d

```diff
@@ -2502,7 +2502,7 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
 
         // Eval UDA in this same scope. Issues 19344, 20835, 21122
         if (em.userAttribDecl)
-            em.userAttribDecl.setScope(sc);
+            em.userAttribDecl._scope = sc;
```

Comment:
What does this mean for issues 19344, 20835, and 21122 ?

Reply:
setScope does some critical things like sc.setNoFree. Skipping it seems very risky.

Reply:
UDAs in enum members are set to eval in the same scope as the enum member, this was the best solution that I found for the issues because the way enums are optimized in the AST.

Reply:
> setScope does some critical things like sc.setNoFree. Skipping it seems very risky.

I really don't know `setNoFree`.
But I think that shouldn't affect any of the UDAs elements internal scope if they had any.

Reply:
Walter you're right. I've added `sc.setNoFree();` because the scope can be used after semantic.
I realized that GDC and LDC call getAttributes(), which in turn can use the scope.